src/modules/accounting/accounting.py

```diff
@@ -0,0 +1,210 @@
+        logger.info({'msg': 'Check if contract could accept report.', 'value': is_reportable})
+        return is_reportable
+
+    @lru_cache(maxsize=1)
+    def _get_processing_state(self, blockstamp: BlockStamp) -> ProcessingState:
+        ps = named_tuple_to_dataclass(
+            self.report_contract.functions.getProcessingState().call(block_identifier=blockstamp.block_hash),
+            ProcessingState,
+        )
+        logger.info({'msg': 'Fetch processing state.', 'value': ps})
+        return ps
+
+    def is_extra_data_submitted(self, blockstamp: BlockStamp) -> bool:
+        processing_state = self._get_processing_state(blockstamp)
+        return processing_state.extra_data_items_count == processing_state.extra_data_items_submitted
+
+    def _calculate_report(self, blockstamp: BlockStamp) -> ReportData:
+        validators_count, cl_balance = self._get_consensus_lido_state(blockstamp)
+
+        finalization_share_rate = self._get_finalization_shares_rate(blockstamp)
+        last_withdrawal_id_to_finalize = self._get_last_withdrawal_request_to_finalize(blockstamp)
+
+        exited_validators = self.lido_validator_state_service.get_lido_new_exited_validators(blockstamp)
+
+        # Here report all exited validators even they were reported before.
+        if exited_validators:
+            staking_module_id_list, exit_validators_count_list = zip(*exited_validators.items())
+        else:
+            staking_module_id_list = exit_validators_count_list = []
+
+        extra_data = self.lido_validator_state_service.get_extra_data(blockstamp, self._get_chain_config(blockstamp))
+
+        # Filter stuck and exited validators that was previously reported
+        report_data = ReportData(
+            consensus_version=self.CONSENSUS_VERSION,
+            ref_slot=blockstamp.ref_slot,
+            validators_count=validators_count,
+            cl_balance_gwei=cl_balance,
+            stacking_module_id_with_exited_validators=staking_module_id_list,
+            count_exited_validators_by_stacking_module=exit_validators_count_list,
+            withdrawal_vault_balance=self._get_withdrawal_balance(blockstamp),
+            el_rewards_vault_balance=self._get_el_vault_balance(blockstamp),
+            last_withdrawal_request_to_finalize=last_withdrawal_id_to_finalize,
+            finalization_share_rate=finalization_share_rate,
+            is_bunker=self._is_bunker(blockstamp),
+            extra_data_format=extra_data.format,
+            extra_data_hash=extra_data.data_hash,
+            extra_data_items_count=extra_data.items_count,
+        )
+
+        return report_data
+
+    @lru_cache(maxsize=1)
+    def _get_consensus_lido_state(self, blockstamp: BlockStamp) -> tuple[int, Gwei]:
+        lido_validators = self.w3.lido_validators.get_lido_validators(blockstamp)
+
+        count = len(lido_validators)
+        balance = Gwei(sum(int(validator.validator.balance) for validator in lido_validators))
```

Comment:
I would suggest renaming to improve readability, smth like this
`count` -> `ever_deposited_count`
`balance` -> `total_balance`

and it's worth adding a comment that the contract expects all validators, including those who exited or slashed